recipient-resolver/src/main/java/com/redhat/cloud/notifications/recipientresolver/router/pojo/RecipientQuery.java

```diff
@@ -0,0 +1,58 @@
+package com.redhat.cloud.notifications.recipientresolver.router.pojo;
+
+import com.redhat.cloud.notifications.recipientresolver.model.RecipientSettings;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.util.Set;
+
+public class RecipientQuery {
+    @NotNull
+    String orgId;
+    @NotNull
+    Set<RecipientSettings> recipientSettings;
+    Set<String> subscribers;
+    boolean isOptIn;
+
+    @Min(value = 0, message = "The offset must be positive")
+    int offset = 0;
```

Comment:
As I wrote in another comment, I don't think will use pagination in this module.
```suggestion
```